http_parser.cc

```diff
@@ -0,0 +1,104 @@
+#include "http_parser.h"
+#include <string>
+#include <vector>
+#include <boost/algorithm/string/split.hpp>
+#include <boost/algorithm/string.hpp>
+#include <boost/regex.hpp>
+
+HttpParser* HttpParser::MakeHttpParser(const char* const raw_req)
```

Comment:
Just tested with setting the max length to 10 instead, and it seems to still work fine